test/validateJoi.js

```diff
@@ -0,0 +1,34 @@
+import chai, { expect } from 'chai';
+import dirtyChai from 'dirty-chai';
+import asPromised from 'chai-as-promised';
+
+import validate from '../src/validateJoi';
+
+chai.use(asPromised);
+chai.use(dirtyChai);
+
+describe('joi validation', () => {
+  context('when the schema is not valid', () => {
+    const schema = {
+      validate: (value, callback) => {
+        const err = new Error();
+        err.name = 'ValidationError';
+
+        return callback(err);
+      },
+    };
+
+    it('rejects', () => expect(validate(schema, null)).to.be.rejected());
+
+    it('sets the error code to 400', () => validate(schema, null)
```

Comment:
I suppose the wording is ambiguous depending on where you read a description. I've always understood it to be:

`400` - something is fundamentally broken in the request, e.g. invalid request message framing
`422` - "generally everything looks good and I can read the body ok, but the body is malformed vs what I expect"
`409` - "the request is correctly formed and I would love to process it but cannot due to the current state of the application"